libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ParsedMediaType.java

```diff
@@ -0,0 +1,134 @@
+     * @throws IllegalArgumentException if the header is malformed
+     */
+    public static ParsedMediaType parseMediaType(String headerValue) {
+        if (DEFAULT_ACCEPT_STRING.equals(headerValue) || "*/*".equals(headerValue)) {
+            return null;
+        }
+        if (headerValue != null) {
+            final String[] elements = headerValue.toLowerCase(Locale.ROOT).split("[\\s\\t]*;");
+
+            final String[] splitMediaType = elements[0].split("/");
+            if ((splitMediaType.length == 2 && TCHAR_PATTERN.matcher(splitMediaType[0].trim()).matches()
+                && TCHAR_PATTERN.matcher(splitMediaType[1].trim()).matches()) == false) {
+                throw new IllegalArgumentException("invalid media type [" + headerValue + "]");
+            }
+            if (elements.length == 1) {
+                return new ParsedMediaType(splitMediaType[0].trim(), splitMediaType[1].trim(), Collections.emptyMap());
+            } else {
+                Map<String, String> parameters = new HashMap<>();
+                for (int i = 1; i < elements.length; i++) {
+                    String paramsAsString = elements[i].trim();
+                    if (paramsAsString.isEmpty()) {
+                        continue;
+                    }
+                    String[] keyValueParam = elements[i].trim().split("=");
+                    if (keyValueParam.length == 2) {
+                        String parameterName = keyValueParam[0].toLowerCase(Locale.ROOT).trim();
+                        String parameterValue = keyValueParam[1].toLowerCase(Locale.ROOT).trim();
+                        parameters.put(parameterName, parameterValue);
+                    } else {
+                        throw new IllegalArgumentException("invalid parameters for header [" + headerValue + "]");
+                    }
+                }
+                return new ParsedMediaType(splitMediaType[0].trim().toLowerCase(Locale.ROOT),
+                    splitMediaType[1].trim().toLowerCase(Locale.ROOT), parameters);
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Resolves this instance to a MediaType instance defined in given MediaTypeRegistry.
+     * Performs validation against parameters.
+     * @param mediaTypeRegistry a registry where a mapping between a raw media type to an instance MediaType is defined
+     * @return a MediaType instance
+     */
+    public  <T extends MediaType> T toMediaType(MediaTypeRegistry<T> mediaTypeRegistry) {
+        T type = mediaTypeRegistry.typeWithSubtypeToMediaType(mimeTypeWithoutParams());
+        if (type != null) {
+
+            Map<String, Pattern> registeredParams = mediaTypeRegistry.parametersFor(mimeTypeWithoutParams());
+            for (Map.Entry<String, String> givenParamEntry : parameters.entrySet()) {
+                if (isValidParameter(givenParamEntry.getKey(), givenParamEntry.getValue(), registeredParams) == false) {
+                    return null;
```

Comment:
good point.. somehow I thought that unknown parameters should not be allowed.
But so far we allowed this and we would succeed in parsing.
This should be fixed by https://github.com/elastic/elasticsearch/issues/63080

Reply:
The concern here is that someone requests application/yaml;charset=Windows-1252, they will now get back application/json (since it is the null/default), right ? 

I think this should log deprecation warning about an unsupported charset and still return the `type`.

Reply:
good point.. how about we remove the validation in this PR?

Reply:
I have removed charset validation from this PR. We have an infrastructure to do this with this PR, but it should be enabled in a separate PR just to make it more focused. I added a note here #63080